src/csharp/Adapters.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.ML.OnnxRuntimeGenAI;
+using System;
+using System.Runtime.InteropServices;
+
+/// <summary>
+/// A container of adapters.
+/// </summary>
+public class Adapters : SafeHandle
```

Comment:
When should one use SafeHandle over IDisposable?

Reply:
It's complicated. The latest I heard it is OK to use it.